src/orion/client/__init__.py

```diff
@@ -338,9 +338,7 @@ def workon(
             max_broken=max_broken,
         )
 
-        producer = Producer(experiment)
-
-        experiment_client = ExperimentClient(experiment, producer)
+        experiment_client = ExperimentClient(experiment)
```

Comment:
Yes exactly! I found the same bug thanks to type hints on my end yesterday:
![potential_bug](https://user-images.githubusercontent.com/13387299/177842695-3caf27de-b04d-4685-aeb7-1cc72b5dcaef.png)